File: even_odd_without_modulo.swift
// http://codegolf.stackexchange.com/questions/2303/how-to-determine-if-a-number-is-odd-or-even-without-mod-or-bitwise-operations

func eo(n:Int) -> String {
  if n == 1 {
    return "odd"
  } else if n == 0 {
    return "even"
  } else {
    return eo(n-2)
  }
}

println(eo(123))

File: even_odd_without_modulo.rb
# http://codegolf.stackexchange.com/questions/2303/how-to-determine-if-a-number-is-odd-or-even-without-mod-or-bitwise-operations

def e_o?(n)
  n == 1 ? "odd" : n == 0 ? "even" : e_o?(n-2)
end

(1..25).each do |n|
  puts "#{n}: #{e_o?(n)}"
end
